File: src/mysqlcookbook/recipes/metadata/get_connection_meta.py
#!/usr/bin/python3
# get_connection_meta.py: get connection metadata

import sys
import mysql.connector
import cookbook

try:
  conn = cookbook.connect()
except mysql.connector.Error as e:
  print("Error code: %s" % e.errno)
  print("Error message: %s" % e.msg)
  sys.exit(1)

try:
#@ _CURRENT_DATABASE_
  cursor = conn.cursor()
  cursor.execute("SELECT DATABASE()")
  row = cursor.fetchone()
  cursor.close()
  if row is None or len(row) == 0 or row[0] is None:
    db = "(no database selected)"
  else:
    db = row[0]
  print("Default database: %s" % db)
#@ _CURRENT_DATABASE_
except mysql.connector.Error as e:
  print("Error code: %s" % e.errno)
  print("Error message: %s" % e.msg)
  sys.exit(1)

conn.close()

File: src/mysqlcookbook/recipes/metadata/get_connection_meta.rb
#!/usr/bin/ruby
# get_connection_meta.rb: get connection metadata

require "Cookbook"

begin
  client = Cookbook.connect
rescue Mysql2::Error => e
  puts "Could not connect to server"
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
end

#@ _CURRENT_DATABASE_
db = client.query("SELECT DATABASE()").first.values[0]
puts "Default database: " + (db.nil? ? "(no database selected)" : db)
#@ _CURRENT_DATABASE_

client.close
